src/org/zaproxy/zap/extension/pscanrulesAlpha/CacheableScanner.java

```diff
@@ -407,7 +407,17 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 								lifetimeFound = true;
 								lifetimesFound++;
 								//get the portion of the string after "maxage="
-								lifetime = Long.parseLong(directiveToken.substring("max-age=".length()));
+								// Split on comma and use 0th item in case there weren't spaces: 
+								// Cache-Control: max-age=7776000,private
+								try {
+									lifetime = Long.parseLong(directiveToken.split(",")[0].substring("max-age=".length()));
+								} catch (NumberFormatException nfe) {
+									lifetimeFound = false;
+									lifetimesFound--;
+									if (logger.isDebugEnabled()) {
+										logger.debug("Could not parse max-age to establish lifetime. Perhaps the value exceeds Long.MAX_VLUE or contains non-number characters:" + directiveToken);
```

Comment:
The typo was not fixed.

Reply:
Doh.....headed for a bus, will fix shortly. Sorry, rushing.